File: Rest-Api-V3/account-and-wallet-private/transfer-post/transfer-post.js
require('dotenv').config();
const CryptoJS = require("crypto-js");
const axios = require('axios');

async function createTransfer(asset, quantity, fromAccount, toAccount) {
  const apiKey = process.env.API_KEY;
  const secretKey = process.env.API_SECRET;
  const ts = new Date().toISOString();
  const nonce = Math.random().toString(36).substring(2);
  const method = "/v3/transfer";
  const apiUrl = "api.ox.fun";

  const body = {
    asset,
    quantity,
    fromAccount,
    toAccount
  };

  // For POST requests, we use the JSON stringified body instead of query params
  const bodyString = JSON.stringify(body);
  const msgString = `${ts}\n${nonce}\nPOST\n${apiUrl}\n${method}\n${bodyString}`;

  const sign = CryptoJS.enc.Base64.stringify(CryptoJS.HmacSHA256(msgString, secretKey));

  const headers = {
    'Content-Type': 'application/json',
    'AccessKey': apiKey,
    'Timestamp': ts,
    'Signature': sign,
    'Nonce': nonce
  };

  try {
    const response = await axios.post(`https://${apiUrl}${method}`, body, { headers });
    if (response.data.success) {
      return response.data.data;
    } else {
      console.error('Failed to create transfer');
    }
  } catch (error) {
    console.error('Error creating transfer:', error.response ? error.response.data : error.message);
  }
}

// Example usage
const asset = 'USDT';
const quantity = '1000';
const fromAccount = '14320';
const toAccount = '15343';

createTransfer(asset, quantity, fromAccount, toAccount).then(data => {
  console.log(data);
});

File: Rest-Api-V3/account-and-wallet-private/transfer-post/transfer-post.py
import os
import requests
import hmac
import hashlib
import base64
import time
import json
from dotenv import load_dotenv

load_dotenv()

def create_transfer(asset, quantity, from_account, to_account):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    ts = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime())
    nonce = str(int(time.time() * 1000))
    method = "/v3/transfer"
    api_url = "api.ox.fun"

    body = {
        "asset": asset,
        "quantity": quantity,
        "fromAccount": from_account,
        "toAccount": to_account
    }

    # For POST requests, we use the JSON stringified body instead of query params
    body_string = json.dumps(body)
    msg_string = f"{ts}\n{nonce}\nPOST\n{api_url}\n{method}\n{body_string}"

    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    headers = {
        'Content-Type': 'application/json',
        'AccessKey': api_key,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    }

    try:
        response = requests.post(f"https://{api_url}{method}", json=body, headers=headers)
        response.raise_for_status()
        data = response.json()
        if data.get('success'):
            return data['data']
        else:
            print('Failed to create transfer')
    except requests.exceptions.RequestException as error:
        print('Error creating transfer:', error)

# Example usage
asset = 'USDT'
quantity = '1000'
from_account = '14320'
to_account = '15343'

response = create_transfer(asset, quantity, from_account, to_account)
print(response)
